src/main/java/org/numenta/nupic/network/KryoSerializer.java

```diff
@@ -54,15 +48,15 @@ private void initFST() {
      */
     @Override
     public void write(Kryo kryo, Output output, T t) {
-        FSTObjectOutput writer = fastSerialConfig.getObjectOutput(output);
         try {
             if(t instanceof Persistable) {
                 ((Persistable) t).preSerialize();
             }
-
-            writer.writeObject(t, t.getClass());
+            
+            byte[] bytes = fastSerialConfig.asByteArray(t);
```

Comment:
Please use the `FSTObjectOutput` interface, rather than the convenience methods on `FSTConfiguration` which have ugly internals.

Avoid this byte array copy by using the streaming approach.